### PR TITLE
ssm-session-manager: Build from source; 1.2.331.0 -> 1.2.398.0 

### PR DESCRIPTION
**Pull request by amarshall**
###### Description of changes

Previously this package was marked unfree binary, now it is a free-licensed source build. Additionally, update to latest version.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix [hunk 1]
```diff
@@ -1,51 +1,37 @@
...
+  '';
 
   installPhase = ''
-    runHook preInstall
-
-    install -m755 -D ${archivePath}/bin/session-manager-plugin $out/bin/session-manager-plugin
-
-    runHook postInstall
+    mkdir -p $out/bin
+    cp "$NIX_BUILD_TOP/go/bin/sessionmanagerplugin-main" $out/bin/session-manager-plugin
```
Comment (NickCao) [inline pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix:26]: Use of `$NIX_BUILD_TOP` is discouraged: https://github.com/NixOS/nixpkgs/issues/189691

File: pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix [hunk 2]
```diff
@@ -1,51 +1,37 @@
...
+  '';
 
   installPhase = ''
-    runHook preInstall
-
-    install -m755 -D ${archivePath}/bin/session-manager-plugin $out/bin/session-manager-plugin
-
-    runHook postInstall
+    mkdir -p $out/bin
+    cp go/bin/sessionmanagerplugin-main $out/bin/session-manager-plugin
```
Comment (NickCao) [inline pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix:26]:
```suggestion
    runHook preInstall
    install -Dm555 go/bin/sessionmanagerplugin-main "$out/bin/session-manager-plugin"
    runHook postInstall
```

File: pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix [hunk 3]
```diff
@@ -1,51 +1,37 @@
...
+    repo = "session-manager-plugin";
+    rev = version;
+    sha256 = "MnTiczFdiP9Xn9fkElVp6tPl7K7PWNyOGjsyLnQHreY=";
+  };
 
-  unpackPhase = if stdenv.isDarwin then "unzip $src" else "dpkg-deb -x $src .";
+  postPatch = ''
+    mv vendor{,-old}
+    mv vendor-old/src vendor
+    rm -r vendor-old
```
Comment (NickCao) [inline pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix:21]: Also you should write the version number into `VERSION`, so `session-manager-plugin --version` works correctly.

**Review by NickCao — COMMENTED** [review]
Apart from the nitpicks, no blockers for me.

**Review by martinbaillie — APPROVED** [review]
Lgtm. Feel free to add to the maintainers list as well.

**Comment by NickCao** [comment]
@ofborg eval

**Pull request merged**